### PR TITLE
Add API to upload files to s3

### PR DESCRIPTION
**Pull request by himani39**
Formio open source version does not support this feature out of box. So we have added the endpoint to support the file upload feature in form submissions.
This API will return the signature for CORS upload to s3 in the format expected by formio.js

#https://github.com/quintype/itsman/issues/6732

### CODE REVIEW

File: .vscode/launch.json [hunk 1]
```diff
@@ -0,0 +1,17 @@
+{
+    // Use IntelliSense to learn about possible attributes.
+    // Hover to view descriptions of existing attributes.
+    // For more information, visit: https://go.microsoft.com/fwlink/?linkid=830387
```
Comment (deoxilix) [inline .vscode/launch.json]: consider removing the comments.

File: docker-compose.yml [hunk 2]
```diff
@@ -40,9 +40,12 @@ services:
       - ./:/app:rw
     environment:
       DEBUG: formio:*
-      NODE_CONFIG: '{"mongo": "mongodb://mongo:27017/formio"}'
+      NODE_CONFIG: '{"mongo": "mongodb://mongo:27017/formio-1"}'
```
Comment (deoxilix) [inline docker-compose.yml]: is there a reason why we suffixed `-1`?

File: src/resources/FileUpload.js [hunk 3]
```diff
@@ -0,0 +1,75 @@
...
+  const key = req.hostname;
+
+  // Get AWS credentials
+  const awsConfig = new AWS.Config();
+
+  const config = {
+    bucket: process.env.AWS_BUCKET,
+    accessKey: awsConfig.credentials.accessKeyId,
+    secretKey: awsConfig.credentials.secretAccessKey,
+  };
```
Comment (deoxilix) [inline src/resources/FileUpload.js]: consider moving this out of the function.
could be done when there's another function requires the data as well.

File: src/resources/FileUpload.js [hunk 4]
```diff
@@ -0,0 +1,75 @@
...
+
+  // Generate policy
+  const policy = generatePolicy(fileName, fileType, config.bucket, key);
+
+  // Sign the base64 encoded policy
+  const signature = generateSignature(config.secretKey, policy);
```
Comment (deoxilix) [inline src/resources/FileUpload.js]: Any reason why we're not using the [SDK: createPresignedPost](https://docs.aws.amazon.com/AWSJavaScriptSDK/latest/AWS/S3.html#createPresignedPost-property) ?

**Review by deoxilix — CHANGES_REQUESTED** [review]

**Review by deoxilix — APPROVED** [review]

File: src/resources/FileUpload.js (hunk 4 above)
Comment (shriram-ethiraj) [reply src/resources/FileUpload.js]: Using SDK:createPresignedPost now to sign file

**Pull request merged**